lib/mxlib/matrix.js

```diff
@@ -322,7 +319,19 @@ var setJoinRule = function(client, roomId, joinRule) {
         roomId, "m.room.join_rules", {
             join_rule: joinRule
         }, ""
-    );
+    ).then(function() {
+        return roomId;
+    });
+};
+
+var setRoomHistoryVisibility = function(client, roomId, history_visibility) {
+    return client.sendStateEvent(
+        roomId, "m.room.history_visibility", {
+            "history_visibility": history_visibility
+        }, ""
+    ).then(function() {
+        return roomId;
```

Comment:
Why do we return roomId here?  Surely we already know it and nothing's happened that would cause it to change?